File: src/main/java/com/homeclimatecontrol/esphome2influxdb/MqttEndpoint.java
package com.homeclimatecontrol.esphome2influxdb;

/**
 * MQTT broker endpoint.
 */
public class MqttEndpoint extends Endpoint {

    /**
     * Topic filter.
     *
     * This filter will be passed to {@link MqttReader}, unlike the filters in
     * {@link Device#topicPrefix} which will be applied locally.
     */
    public String topic = "#";

    /**
     * Whether to reconnect automatically.
     *
     * Automatic reconnect is disabled by default, here's why:
     * https://github.com/hivemq/hivemq-mqtt-client/issues/496
     */
    public Boolean autoReconnect = false;

    public MqttEndpoint() {
        setPort(1883);
    }

    @Override
    protected void render(StringBuilder sb) {

        super.render(sb);

        sb.append(",topic=").append(topic);
    }
}

File: src/main/kotlin/com/homeclimatecontrol/esphome2influxdb/k/MqttEndpoint.kt
package com.homeclimatecontrol.esphome2influxdb.k

/**
 * MQTT [Endpoint].
 *
 * @param topic Topic filter. This filter will be passed to [MqttReader], unlike the filters in [Device.topicPrefix] which will be applied locally.
 */
class MqttEndpoint(
    var topic: String = "#"
) : Endpoint() {

    init {
        setPort(1883)
    }

    override fun render() = super.render().plus(",topic=$topic")
}
